### PR TITLE
Feature update workflow infra

### LINKED ISSUE #253: Configuration changes should trigger rerun of the workflow
### What can be improved?

Currently, rules are not rerun when the configuration of the workflow changes. This is a problem for example when running the minimal workflow or more generally, when changing the configuration.

This is implemented in Snakemake >= 7.8.

### Version

1.2.0.dev

### LINKED ISSUE #255: Simplify the Snakemake call
The profile mechanism right now is clunky, as the profile must be defined in each call to Snakemake. Instead, we can and should set the profile in the conda environment automatically.

See https://github.com/timtroendle/cookiecutter-reproducible-research/issues/14 for an implementation.

This requires Snakemake >= 6.5.0.

### PR DESCRIPTION
**Pull request by timtroendle**
Fixes #253

Fixes #255 

## Checklist

Any checks which are not relevant to the PR can be pre-checked by the PR creator. All others should be checked by the reviewer. You can add extra checklist items here if required by the PR.

- [x] CHANGELOG updated
- [x] Minimal workflow tests pass
- [x] Tests added to cover contribution
- [x] Documentation updated
- [x] Configuration schema updated

### CODE REVIEW

**Review by brynpickering — CHANGES_REQUESTED** [review]
Nice!

I'm just running the minimal workflow locally, to double check it ;)

docs env file needs fixing, see my separate comment on that.

File: envs/docs.yaml [hunk 1]
```diff
@@ -13,4 +13,4 @@ dependencies:
     - pip=21.0.1
     - pip:
         - jsonschema2md==0.2.1
-        - -e ../lib[docs]
+        - -e ./lib[docs]
```
[the remark this answers is not in the document]
Comment (timtroendle) [reply envs/docs.yaml:16]: Nice catch. I moved it to the top level.

**Comment by timtroendle** [comment]
I have no clue why the environment cannot be resolved on readthedocs. It works just fine on my machine.

It maybe a Linux problem, but I cannot test that right now as Euler is down for maintenance. Then again, the missing dependency `toposort>=1.10` is a `noarch` package, so it does not really make sense to me.

**Comment by brynpickering** [comment]
You can always trust mamba to solve all your issues :)

**Review by brynpickering — APPROVED** [review]

**Pull request merged**